### ISSUE TITLE
Again having issue sharing to EinkBro from other Android applications/browsers

### ISSUE THREAD

**Issue by CottonEvoker** [plateaukao/einkbro#353]
> @CottonEvoker 
> You could try enable Settings > Behaviour > Trim input text content. 
> 
> The reason it happens is that Palabre does not just share url. It also contains title in the content. So, its not recognized as a valid url in EinkBro.
> 
> _Originally posted by @plateaukao in https://github.com/plateaukao/einkbro/discussions/300#discussioncomment-7874562_

Hi again Daniel,

I come back to you about the issue I raised on thread #300. 

This problem (with link sharing to EinkBro) is occurring again, but in a broader way than before, i.e. not impacting only shares to EinkBro from Palabre, but from other applications/browsers, and I'm unable to resolve it using the remedy you suggested on #300.

When I share a link (e.g. to a news story) to EinkBro using the standard Android share menu, instead of the link opening in EinkBro, the page that opens in EinkBro is a search in DuckDuckGo of the url, requiring that I then tap on the result (if it's showing) in the DDG search results to view the story.

This happens despite having "Trim input text content" selected, and I'm encountering it now not just when when sharing to EinkBro from Palabre, but from any Android application or other browser.

I attached two PDFs as separate examples. Each contains screenshots showing what happens when sharing links from the Samsung Internet browser to EinkBro. (The result is exactly the same though if I share from any other browser like Edge or Chrome). 

"Example 1.pdf"

* Here, I share a link to the Wired.com story "What Ever Happened to the Tiny House Movement?", i.e. url https://www.wired.com/story/what-ever-happened-to-the-tiny-house-movement.
* In the page that opens in EinkBro, this is searched in DuckDuckGo as "https%3A%2F%2Fwww.wired.com%2Fstory%2Fwhat-ever-happened-to-the-tiny-house-movement%2F"
* In this case though, the actual link to the story is the first match in the search results, so I can at least click on that.

"Example 2.pdf"

* Here, I share a link to story on theguardian.com, "AI likely to increase energy use and accelerate climate misinformation – report", url https://www.theguardian.com/technology/2024/mar/07/ai-climate-change-energy-disinformation-report.
* In the page that opens in EinkBro, this is searched as "https%3A%2F%2Fwww.theguardian.com%2Ftechnology%2F2024%2Fmar%2F07%2Fai-climate-change-energy-disinformation-report"
* But in this case, unlike in Example 1, the actual link to the story does not show in the search results, so that I can only click theguardian.com and try to click through to the story.

For your information: I don't know if it's relevant, but when I contacted you in November on #300, I was then using the Play Store version of EinkBro. After they removed it, I uninstalled the Play Store version and installed the Github version. I can't confirm definitively though that is is only since switching to the Github version (of which I currently have v. 11.5.0 installed) that I've been encountering this share issue again.

I hope this can be resolved. Thank you in advance for any time you take to look at it.

[Example 1.pdf](https://github.com/plateaukao/einkbro/files/14538602/Example.1.pdf)
[Example 2.pdf](https://github.com/plateaukao/einkbro/files/14538603/Example.2.pdf)

**Comment by CottonEvoker** [comment]
I'm getting it on all three Android devices I use Einkbro on.

I tested just now sharing from each of Samsung Internet, Chrome & Edge to Einkbro on all three devices and exactly the same things happens in all cases.

Main device, on which I took the screenshots for the Example 1 & 2 attachments, is a OnePlus Nord 2T 5G running OxygenOS 14. The other two are a Lenovo Yoga Tab 11 running Android 12 and an older backup phone, Sony Xperia XZ2 Compact running Android 10.

**Comment by plateaukao** [comment]
could you let me know what search settings you have configured?
maybe specific combination caused this issue.

**Comment by CottonEvoker** [comment]
Attaching screenings from EinkBro settings on the OnePlus Nord 2T 5G, but search settings the same too on the a Lenovo Yoga Tab and the Sony Xperia XZ2 Compact.

![Search engine](https://github.com/plateaukao/einkbro/assets/98884661/2b1866ae-d46a-4e86-a108-887c1315d08c)
![Search settings](https://github.com/plateaukao/einkbro/assets/98884661/6a553341-e53c-4aa4-a8b5-a7b159c2fd8c)

**Comment by plateaukao** [comment]
please update to 11.6.0 to see if it helps. recently I refactor codes related to external search.

**Comment by CottonEvoker** [comment]
Thank you Daniel, that did solve the sharing problem.

But I'm encountering an issue or 'side effect' with this new version: My phone now no longer seems to recognise EinkBro as a browser - EinkBro no longer shows as one of the options under "Default browser app" under "Default apps".

Also, on my Android homescreen I keep several folders, each containing website shortcuts to open in a particular browser. I have one folder for Chrome, another for Samsung Internet and one for EinkBro. I created all the website shortcuts in Tasker using Send Intent tasks. But since version 11.6.0, none of these links will now open in EinkBro.

I thought at first that perhaps the package name for EinkBro (info.plateaukao.einkbro) changed with 11.6.0, that this might be why my Tasker website shortcut tasks no longer work. (In the Tasker Send Intent task, you let Tasker know what browser you want the link sent to by inputting the package name, i.e. com.android.chrome for Chrome or com.sec.android.app.sbrowser for Samsung Internet). So I thought that modifying the package name in the Tasker tasks would fix it, but I've just checked and the EinkBro package name hasn't changed on 11.6.0. 

I'm pleased to get sharing going again, but do you have any idea what can be behind these changes on 11.6.0? I'd like to be able to have the option to set EinkBro as my default browser again and to have my EinkBro Tasker website shortcuts work again.

By the way, I cleared cache and data in EinkBro v11.6.0, but EinkBro still doesn't show as a selectable Default browser app, and nor either do the Tasker website shortcuts work. I also completely uninstalled EinkBro and reinstalled v11.6.0, but again without effect.

**Comment by CottonEvoker** [comment]
"_should have been fixed in v11.6.1_"

I'm pleased to report it certainly is!

Thank you so much. I have EinkBro in peak performance, and will update it to 11.6.1 on my other devices now 😊 🙏

FYI: For the first time, whilst updating to v11.6.1, AVG AntiVirus flagged it as a suspicious app: "Suspicious app detected: EinkBro has been reported as a suspicious app". I was able to proceed by tapping the warning's Ignore option.
